opt_adapt/opt.py

```diff
@@ -41,6 +41,7 @@ def __init__(self, options={}):
         self.model_options = {}
         self.disp = 0
         self.lr = 0.001  # Step length / learning rate
+        self.lr_lowerbound = 1e-25
```

Comment:
This is smaller than machine precision, so even slightly larger numbers would probably result in division by zero errors. I think 1e-8 would probably be more appropriate.

Reply:
This will not necessarily always be the case, though, so I think 1e-8 would be safer.

Reply:
Doctor, If I could make a change that only BFGS and LBFGS methods need to check this condition? So that the other three methods don't need to be influenced.

Reply:
I found that for Adam, it quite often that lr to less than 1e-15 by line search.

Reply:
You could add a parameter to `OptAdaptParameters` that says whether or not to do this check. It could default to `True` for BFGS and L-BFGS and default to `False` otherwise?

Reply:
Doctor, I am wondering how could I set the different defaults according to the method?

Reply:
Oh for that you will need the PR that we merged yesterday.